src/genie/libs/parser/iosxe/show_redundancy.py

```diff
@@ -0,0 +1,94 @@
+import re
+
+from genie.metaparser import MetaParser
+from genie.metaparser.util.schemaengine import Any, Optional
+
+
+# =======================================
+# Schema for:
+#  * 'show redundancy switchover history'
+# =======================================
+class ShowRedundancySwitchoverHistorySchema(MetaParser):
+    """Schema for show redundancy switchover history."""
+
+    schema = {
+        Optional("index"): {
+            Any(): {
+                "current_active": int,
+                "previous_active": int,
+                "switchover_reason": str,
+                "switchover_time": str,
+            },
+        }
+    }
+
+
+# =======================================
+# Parser for:
+#  * 'show redundancy switchover history'
+# =======================================
+class ShowRedundancySwitchoverHistory(ShowRedundancySwitchoverHistorySchema):
+    """Parser for show redundancy switchover history"""
+
+    cli_command = ['show redundancy switchover history']
+
+    def cli(self, output=None):
+        if output is None:
+            output = self.device.execute(self.cli_command[0])
+
+        # Index  Previous  Current  Switchover             Switchover
+        #        active    active   reason                 time
+        # -----  --------  -------  ----------             ----------
+        #    1       1        2     Active lost GW         03:26:09 UTC Wed Sep 23 2020
+        #    2       2        1     user forced            05:40:13 UTC Thu Oct 1 2020
+
+
+        #    1       1        2     Active lost GW         03:26:09 UTC Wed Sep 23 2020
+        switchover_history_capture = re.compile(
+            # 1
+            r"^(?P<index>\d+)\s+"
+            # 1
+            r"(?P<previous_active>\d+)\s+"
+            # 2
+            r"(?P<current_active>\d+)\s+"
+            # Active lost GW
+            r"(?P<switchover_reason>.+)\s+"
+            # 03:26:09 UTC Wed Sep 23 2020
+            r"(?P<switchover_time>\d+:\d+:\d+\sUTC\s\S+\s\S+\s\d+\s\d+)$"
+        )
+
+        redundancy_info_obj = {}
+
+        for line in output.splitlines():
+            line = line.strip()
+
+            if switchover_history_capture.match(line):
+                match = switchover_history_capture.match(line)
+                group = match.groupdict()
+
+                # cleanup dict
+                for item in group:
+                    # remove trailing spaces from regex capture
+                    group[item] = group[item].strip()
+
+                    # convert str to int
+                    try:
+                        group[item] = int(group[item])
+                    except ValueError:
```

Comment:
What condition does this happen in?

Reply:
If the `str` can be turned into an `int`. So `"1"` will changed to `1` but `"some string"` stays `"some string"`. I converted them one by one before, but this made more sense to me

Reply:
So as an example, if `switchover_reason` was an error code, such as 3557, it would be cast into an integer and fail on schema validation?

Reply:
Very true. I'll redo it to keep the schema validation in mind